R/sce_to_anndata.R

```diff
@@ -60,10 +60,35 @@ sce_to_anndata <- function(
   # assign SCE to new variable to avoid modifying input SCE
   sce_to_convert <- sce
 
-  # remove any objects or dataframes
+  # keep only vectors and data frames; no objects or DataFrames
   metadata_to_keep <- metadata(sce_to_convert) |>
-    purrr::discard(is.object) |>
-    purrr::discard(is.list)
+    purrr::keep(\(x) {
+      is.atomic(x) || is.data.frame(x)
+    })
+
+
+  # prepare data frames for anndata conversion:
+  # - remove any list columns
+  # - ensure any NA values in character columns are formatted as character
+  # - ensure any
+  metadata_to_keep <- metadata_to_keep |>
+    purrr::map(
+      \(meta) {
+        if (is.data.frame(meta)) {
+          meta <- meta |>
+            dplyr::select(dplyr::where(\(col) !is.list(col))) |>
+            dplyr::mutate(
+              dplyr::across(dplyr::where(is.character), \(col) tidyr::replace_na(col, "NA")),
```

Comment:
Do we need to do this for character vectors too?

Reply:
Yes - if there are any NAs in a character column, the conversion can't handle multiple types and just fully fails.
Possibly we want this kind of code later in this function where we make sure coldata/rowdata are ok, too.

Reply:
Sorry, I meant any vectors in the metadata list itself. I don't see any place where this will happen, but it could. 

I am not concerned about `coldata` and `rowdata`; I expect that zellkonverter is handling this internally.

Reply:
Character vectors with `NA` work fine; this conversion "bug" appears to just affect data frames